### PR TITLE
API development guidelines

### PR DESCRIPTION
**Pull request by targos**
I'm trying to formalize how we should implement the public APIs in ml.js.

@mljs/collaborators Please take a look !

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+### new Predictor()
+
+Creates the predictor. The constructor can take parameters or options to initialize the algorithm.
+
+### predictor.train(trainingSet)
+
+This method is optional.  
+If the predictor has a training phase, it is executed here.
+
+### predictor.predict(data)
```
Comment (targos) [inline doc/API.md]: currently we have different names for this one depending on the algorithm. For example it's `project` in PCA or `compute` in the regression.
I think it is easier if we always have `predict`.

**Comment by targos** [comment]
If you have ideas about other topics to add in this file, don't hesitate to tell me

**Comment by JeffersonH44** [comment]
it would be nice to have a score method, no?

**Comment by targos** [comment]
@JeffersonH44 what do you mean ?

**Comment by JeffersonH44** [comment]
a method like this:

score(X, y) and retrieves a score that obviously depends of the predictor, for example the R² score for a regression model

**Comment by targos** [comment]
OK good idea. I'll add it as an optional method.

**Comment by andcastillo** [comment]
About the score, it is usually calculated during the training with the training dataset. Nevertheless, you may be interested in calculating that score for a test set. Also, in some cases the score is more than 1 single value but a set of values.

**Comment by targos** [comment]
Of course, it can be precalculated or contain multiple values. The function can also return different scores depending what parameters are passed to it.
The important thing is that we agree to always call it `score`.

**Comment by maasencioh** [comment]
We also should standarize if the input is [x1,y1][x2,y2] or [x1,x2][y1,y2]....

**Comment by targos** [comment]
Absolutely.
I will open a new PR for that and another thing I have in mind (platform compatibility).
